File: src/Best Time to Buy and Sell Stock II.cpp
class Solution {
public:
    int maxProfit(vector<int> &prices) {
        // Note: The Solution object is instantiated only once and is reused by each test case
        if(prices.size() == 0)
            return 0;
        int ans = 0;
        int firstbuy = -1;
        for(int i=0; i<prices.size()-1; i++)
        {
            if(prices[i+1] > prices[i])
            {
                firstbuy = i;
                break;
            }
        }
        if(firstbuy == -1)
            return ans;
        int buy = firstbuy;
        int sell = -1;
        while(buy < prices.size())
        {
            int i = buy;
            while(i<prices.size()-1 && prices[i] <= prices[i+1])
                i++;
            sell = i;
            ans += prices[sell] - prices[buy];
            while(i<prices.size()-1 && prices[i] >= prices[i+1])
                i++;
            if(i == prices.size()-1)
                return ans;
            buy = i;
        }
    }
};

File: python/Best Time to Buy and Sell Stock II.py
class Solution:
    # @param prices, a list of integer
    # @return an integer
    def maxProfit(self, prices):
    	goup = False
    	ret = 0
    	for i in range(len(prices)-1):
    		if prices[i+1] >= prices[i] and not goup:
    			buy = prices[i]
    			goup = True
    			continue
    		if prices[i+1] < prices[i] and goup:
    			sell = prices[i]
    			goup = False
    			ret += sell - buy

    	if goup:
    		ret += prices[len(prices)-1] - buy
    	return ret